omgeo/services/__init__.py

```diff
@@ -658,9 +658,11 @@ def _geocode(self, pq):
                 query = dict(query, searchExtent=pq.viewbox.to_esri_wgs_json())
         else: # single-line
             method = 'find'
+            magic_key = pq.key if hasattr(pq, 'key') else ''
```

Comment:
There's some weird behavior with magicKey. We ran into an issue on the CAC project where if you had a request to suggest like so:
```
{
   text: 'somefoo',
   bbox: 'somebbox'
   categories: 'some,multiple,categories'
}
```
and then made a find request afterwards with the magicKey added and the text switched to the user's selection, the find endpoint would fail to match some locations.

If you remove the bbox filter on the find query in that same request, no issues.

Not sure if there are other oddities with passing a magicKey and some of the other filtering params. Just something to watch out for -- to be safe, I would omit the magicKey entirely rather than passing an empty string.

Otherwise, LGTM.